### DISCUSSION TITLE
macOS builds

### DISCUSSION THREAD

**Discussion by pgilfernandez** [friction2d/friction#267] [Ideas]
I was about to try Friction but I got surprised because there is no macOS version.

What is the blocker or just why you don't offer them? Any near future plans about it?

I think it would be a great addition to the final 1.0 version as it would embrace all mainstream OS in the market.

Thanks!

**Comment by rodlie** [comment]
macOS is not supported.

Why?

* There are cache/memory issues on macOS I have not been able to solve, this is a major blocker
* There are many UI issues
* I only have Intel Macs, supporting max macOS 12 (now EOL)
* Maintenance costs (time spent) is more than Windows and Linux combined
  * Also usually the least amount of users (compared to Windows/Linux, based on my experience over the years)

Currently my time is better spent on developing and maintaining Friction for Windows and Linux.

**Reply by pgilfernandez** [reply]
Thanks for the honest answer, I understand your position.

By the way, you are saying that it compiles but it has too many bugs that are difficult to address due to your limited time, is that right? Does it compile?

What if you include some help for brave users/colaborators/devs on how to compile Friction for macOS and open the door for them to help fixing bugs or improving it little by little?
You can always say that you don't support macOS but it could change in the future if the collaboration of third parties rise it to a stable state.

I would have a look at UI ones as an starting point...

Thanks

**Reply by rodlie** [reply]
> > Maintenance costs (time spent) is more than Windows and Linux combined
> 
> You can deploy Mac builds for free using Github CI. Publishing to Apple Store is another thing tho.

Maintaining macOS requires more work, has nothing to do with building.

**Comment by rodlie** [comment]
Public code does not build for macOS anymore (0.9.5 might). 

I have a local repo on one of my Mac's, based on 0.9.6-ish. So it's still possible to make a working build, but due to the cache/memory issue I have not bothered maintaining it.

![Skjermbilde 2024-10-19 kl  00 01 33](https://github.com/user-attachments/assets/14e82e36-aa85-44a4-805c-ef1294ce88d6)


I could publish a 'mac' branch when I get the time, but there will be no official support. This could change in the future, but it's not a high priority from my POV.

**Reply by pgilfernandez** [reply]
It would be great to play with that branch... maybe, who knows, it gets some attention and it gets maintained by other devs.

I'll be looking for it, please drop some words here when you have time to upload it.

Thanks!

**Comment by rodlie** [comment]
So, had some spare time and a macbook on my lap... I was able to get `main` building with various modifications. Also had some time to look at the cache stuff and might have improved the situation (and probably broken Linux). Will upload changes and instructions later this weekend.

![friction-1 0-mac](https://github.com/user-attachments/assets/a6235ea1-7dd4-4610-9614-e6b0f0cc8682)

**Reply by pgilfernandez** [reply]
That's great, I'll stay tuned.

Thanks!

**Reply by tomgaa** [reply]
That sounds awesome! Id also love to try it onmac and maybe even contribute to its development in the future.
I think there's a pretty large user base for graphics programs like After Effects/.Photoshop on Mac.

**Comment by rodlie** [comment]
https://github.com/friction2d/friction/tree/macos

Major blocker (cache/memory) has been fixed (req macOS12+) and several minor UI issues, but it's far from done. Some UI issues must be fixed (mouse/touch input) and the color picker/drop tool is broken (I will try to refactor this tool before 1.0 final).

**Comment by rodlie** [comment]
New issue tracking macOS support: https://github.com/friction2d/friction/issues/272
